Improve descriptions of cmake options (#3878)
nest/nest-simulator: doc/htmldoc/installation/cmake_options.rst

+---------------------------------------------+----------------------------------------------------------------+

> | ``-Dwith-mpi=[OFF|ON]``                     | Build with MPI parallelization [default=OFF].                  |

Reviewer: ```suggestion
| ``-Dwith-mpi=[OFF|ON|</path/to/mpi library>]``                     | Build with MPI parallelization [default=OFF].                  |
```

|                                             | Enables distributed-memory parallel simulation across multiple |
|                                             | processes. Required for ``-Dwith-music`` and                   |

> |                                             | ``-Dwith-sionlib``.                                            |

Reviewer: ```suggestion
|                                             | Optionallly give the directory where an MPI library is installed. Enables distributed-memory parallel simulation across multiple |
|                                             | processes. Required for ``-Dwith-music`` and                   |
|                                             | ``-Dwith-sionlib``.                                            |
```

+---------------------------------------------+----------------------------------------------------------------+

> | ``-Dwith-openmp=[OFF|ON|<OpenMP-Flag>]``    | Build with OpenMP multi-threading [default=ON]. Optionally set |

Reviewer: ```suggestion
| ``-Dwith-openmp=[OFF|ON|</path/to/openmp library>]``    | Build with OpenMP multi-threading [default=ON]. Optionally give |
```

> |                                             | OMP compiler flags. Enables shared-memory multi-threading for  |

Reviewer: ```suggestion
|                                             | the directory where an OpenMP library is installed.  Enables shared-memory multi-threading for  |
```

|                                               | Multi-threaded timers can affect the performance.              |
+-----------------------------------------------+----------------------------------------------------------------+
| ``-Dwith-detailed-timers=[OFF|ON]``           | Build with detailed internal time measurements [default=OFF].  |
|                                               | Detailed timers can affect the performance. Required to enable |
|                                               | ``-Dwith-cycle-timers``.                                       |
+-----------------------------------------------+----------------------------------------------------------------+

> | ``-Dwith-cycle-timers=[OFF|ON]``              | Build with internal per-cycle time measurements                |

Reviewer 2: ```suggestion
| ``-Dwith-cycle-timers=[OFF|ON]``              | Build with internal per-cycle time measurements  and logging of per-cycle spike counts              |
```
